kerastuner/applications/augment.py

```diff
@@ -0,0 +1,303 @@
+            transform_factor_max = transform_params[1]
+            if len(transform_params) > 2:
+                raise ValueError('Length of keyword argument {} must not exceed 2.'
+                                 .format(transform_name))
+        except TypeError:
+            transform_factor_min = 0
+            transform_factor_max = transform_params
+
+        if not (isinstance(transform_factor_max, (int, float)) and
+                isinstance(transform_factor_min, (int, float))):
+            raise ValueError('Keyword argument {} must be int or float, '
+                             'but received {}. '
+                             .format(transform_name, transform_params))
+
+        self.transforms.append((transform_name,
+                                (transform_factor_min, transform_factor_max)))
+
+
+class HyperImageFixedAugment(HyperImageAugment):
+    """An HyperModel for fixed policy augmentation.
+       A tunable hyperparameter is assigned to each of the
+       transforms chosen, and the transforms are applied
+       sequentially.
+       Only supporting augmentations in Keras preprocessing layers.
+
+    # Arguments:
+        input_shape: Optional shape tuple, e.g. `(256, 256, 3)`.
+        input_tensor: Optional Keras tensor (i.e. output of
+            `layers.Input()`) to use as image input for the model.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperImageAugment. See `HyperImageAugment`.
+
+    # Search space:
+        'factor_{transform}' for each of the transforms chosen:
+            Float, range is set through corresponding keyword argument.
+
+    # Usage example:
+        ```
+        hm_aug = HyperImageFixedAugment(input_shape=(32, 32, 3),
+                                        translate_x=0.5,
+                                        translate_y=[0.2, 0.4]
+                                        contrast=0)
+        ```
+        Then the hypermodel `hm_aug` will search 'factor_rotate' in [0, 0.2],
+        'factor_translate_x' in [0, 0.5], 'factor_translate_y' in [0.2, 0.4].
+    """
+
+    def build(self, hp):
+        model = keras.Sequential(name='fixed_augment')
+
+        if self.input_tensor is not None:
+            model.add(self.input_tensor)
+        elif self.input_shape is not None:
+            model.add(keras.Input(shape=self.input_shape))
+
+        for transform, (factor_min, factor_max) in self.transforms:
+            transform_factor = hp.Float(f'factor_{transform}',
+                                        factor_min,
+                                        factor_max,
+                                        step=0.05,
+                                        default=factor_min)
+            if transform_factor == 0:
+                continue
+            transform_layer = TRANSFORMS[transform](transform_factor)
+            model.add(transform_layer)
+
+        return model
+
+
+class HyperImageRandAugment(HyperImageAugment):
```

Comment:
I think instead of having `HyperImageRandAugment` and `HyperImageFixedAugment`, it would be better to have one `HyperImageAugment` class with an optional argument `rangaug_count` that if not set, always applies all transforms

That way users don't have to choose b/t objects, and we can just provide the defaults that we think are best.

WDYT?